app/controllers/boxes_controller.rb

```diff
@@ -100,12 +113,37 @@ def load_box
     end
   end
 
+  def load_box_print
+
+    @box = Box.where(institution: @navigation_context.institution, id: params.fetch(:id)).take
+    if @boxes.nil?
+      @box = Box
+        .joins(box_transfers: :transfer_package)
+        .where("boxes.id = ? and ( transfer_packages.sender_institution_id = ? or transfer_packages.receiver_institution_id = ? )", params.fetch(:id), @navigation_context.institution, @navigation_context.institution).take
+    end
+  end
+
   def load_box_samples
     samples = @box.samples.preload(:batch, :sample_identifiers)
-    samples = samples.scrambled if @box.blinded?
     SamplePresenter.map(samples, request.format)
   end
 
+  def load_box_samples_print
```

Comment:
I did it spliting the print case, because we have an exception for this case. Because the order of the inventory and the printing labels must be equal to the sender. And he must have the same order to both. But he can see the information unblinded on inventory, even when it's blinded. But this situation is only for printing functions.

Reply:
For the receiver yes, but for the sender no. On the beggining I was afraid to use the same method and change something that I can't for other screen. And it's not just the sort, but we have an exception to show blinded data for the sender in Inventory CSV. But I will merge the methods anyway. Now I'm comfortable to do this.